### DISCUSSION TITLE
[question] Differences to `@typescript-eslint/consistent-type-assertions` and `@typescript-eslint/no-non-null-assertion`

### DISCUSSION THREAD

**Discussion by mikew** [Dremora/eslint-plugin-no-type-assertion#6] [Q&A] [answered]
I'm looking to add this rule to my teams linter, and am brushing up on potential other rules to add from `@typescript-eslint`, and noticed it has:

```
'@typescript-eslint/consistent-type-assertions': ['error', { assertionStyle: 'never' }]
```

which disallows type assertions, and:

```
'@typescript-eslint/no-non-null-assertion': ['error']
```

which disallows non-null assertions.

Is the special handling of `unknown` the big differentiator with the rule in this repo?

**Comment by Dremora** [comment, marked as answer]
Pretty much. See also [the thread here](https://github.com/typescript-eslint/typescript-eslint/issues/709#issuecomment-1439055342).
